src/dune_rules/module_compilation.ml

```diff
@@ -8,7 +8,16 @@ open Memo.O
    extension is not .ml or when the .ml and .mli are in different directories.
    This flags makes the compiler think there is a .mli file and will the read
    the cmi file rather than create it. *)
-let force_read_cmi source_file = [ "-intf-suffix"; Path.extension source_file ]
+let force_read_cmi ~obj_dir ~version ~src ~cm_kind module_ =
+  match Version.supports_cmi_file version, Lib_mode.of_cm_kind cm_kind with
```

Comment:
@anmonteiro does melange support `-cmi-file`? I imagine that it does.

Reply:
We don't support `-cmi-file` but I'm not opposed to adding it so we can move towards a single code path in the future.